qutip/solver/ode/krylov.py

```diff
@@ -0,0 +1,237 @@
+from ..integrator import IntegratorException, Integrator
+import numpy as np
+from qutip.core import data as _data
+from scipy.optimize import root_scalar
+from ..sesolve import SeSolver
+
+
+__all__ = ["IntegratorKrylov"]
+
+
+class IntegratorKrylov(Integrator):
+    """
+    Evolve the state vector ("psi0") finding an approximation for the time
+    evolution operator of Hamiltonian ("H") by obtaining the projection of
+    the time evolution operator on a set of small dimensional Krylov
+    subspaces (m << dim(H)).
+    """
+    integrator_options = {
+        'atol': 1e-7,
+        'nsteps': 100,
+        'min_step': 1e-5,
+        'max_step': 1e5,
+        'krylov_dim': 0,
+        'sub_system_tol': 1e-7,
+        'always_compute_step': False,
+    }
+    support_time_dependant = False
+    supports_blackbox = False
+    method = 'krylov'
+
+    def _prepare(self):
+        if not self.system.isconstant:
+            raise ValueError("krylov method only support constant system.")
+        self._max_step = -np.inf
+        self._step = 0
+        krylov_dim = self.options["krylov_dim"]
+        if krylov_dim < 0 or krylov_dim > self.system.shape[0]:
+            raise ValueError("The options 'krylov_dim', must be a positive "
+                             "integer smaller that the system size")
+
+        if krylov_dim == 0:
+            # TODO: krylov_dim, max_step and error (atol) are related by
+            # err ~= exp(-krylov_dim / dt**(1~2))
+            # We could ask for 2 and determine the third one.
```

Comment:
Where does the approximate expression for err come from? I would expect it to depend on the structure of the eigenvalues of the system, and not just the dimension.

Reply:
I tried it with various Hamiltonian and states. The eigenvalues and state affect the constant and power of dt.
However, larger dims decrease the error and `dt` increase it. 
`_compute_max_step` already compute `dt` as a function of `err` for fixed `krylov_dims`. We could compute `krylov_dims` with fixed `dt` and `err` or find optimal `krylov_solve` and `dt`with fixed. But I am not sure it's worth it so I just left a comment that it could be done.